File: src/main/java/persist/BambooPluginSettings.java
package persist;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;


@State(name = "bambooplugin",
        storages = @Storage(file = "bambooplugin.xml"))
public class BambooPluginSettings implements PersistentStateComponent<BambooPluginSettings> {

    public String url;
    public String username;
    public String password;

    @NotNull
    @Override
    public BambooPluginSettings getState() {
        return this;
    }

    @Override
    public void loadState(BambooPluginSettings state) {
        XmlSerializerUtil.copyBean(state, this);
    }

    public static BambooPluginSettings getInstance() {
        return ServiceManager.getService(BambooPluginSettings.class);
    }
}

File: src/main/java/persist/BambooPluginSettings.kt
package persist

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "bambooplugin",
        storages = [(Storage("bambooplugin.xml"))])
data class BambooPluginSettings(var url : String = "",
                                var username : String = "",
                                var password : String = "") : PersistentStateComponent<BambooPluginSettings> {

    override fun getState(): BambooPluginSettings = this

    override fun loadState(p0: BambooPluginSettings?) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        fun getInstance() = ServiceManager.getService(BambooPluginSettings::class.java)
    }
}